### PR TITLE
feature: Add client ip to inbound request information

### PR DESCRIPTION
**Pull request by jimlambrt**

### CODE REVIEW

File: internal/servers/common/client_ip.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+		a = strings.TrimSpace(a)
+		isPrivate, err := isPrivateAddr(ctx, privateNetworks, a)
+		if !isPrivate && err == nil {
+			return a, nil
+		}
+	}
+
+	// finally, just fallback to the contexts of the X-Real-Ip header (which
+	// could be empty)
+	return realIp, nil
```
Comment (louisruch) [inline internal/servers/common/client_ip.go:59]: Would we really want to fall back to an empty `"X-Real-Ip"`?  Wouldn't it be better to fall back to `r.RemoteAddr`
Also if we have `X-Real-Ip` wouldn't that be better than `X-Forwarded-For`?

Seems logic here should be check for `X-Real-Ip` if empty check for `X-Forwarded-For` that is private, if none then fall back to r.RemoteAddr

File: internal/servers/controller/listeners.go [hunk 2]
```diff
@@ -16,12 +16,18 @@ import (
...
 	pbs "github.com/hashicorp/boundary/internal/gen/controller/servers/services"
 	"github.com/hashicorp/boundary/internal/libs/alpnmux"
+	"github.com/hashicorp/boundary/internal/servers/common"
 	"github.com/hashicorp/boundary/internal/servers/controller/handlers/workers"
 	"github.com/hashicorp/go-multierror"
 	"google.golang.org/grpc"
 )
 
 func (c *Controller) startListeners(ctx context.Context) error {
+
```
Comment (louisruch) [inline internal/servers/controller/listeners.go:26]: `make fmt`

**Review by louisruch — COMMENTED** [review]
Need to run `make fmt` also I have one question on logic to get client IP, haven't used the headers too much in the past so going on names to what I think is a potentially better ordering.

File: internal/servers/controller/listeners.go (hunk 2 above)
Comment (jimlambrt) [reply internal/servers/controller/listeners.go:26]: ty!

File: internal/servers/common/client_ip.go (hunk 1 above)
Comment (jimlambrt) [reply internal/servers/common/client_ip.go:65]: ty!

**Review by louisruch — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 3]
```diff
@@ -16,6 +16,8 @@ Canonical reference for changes, improvements, and bugfixes for Boundary.
 
 ### Deprecations/Changes
 
+* controllers: Add client IP to inbound request information which is included in
```
Comment (louisruch) [inline CHANGELOG.md:19]: This should be moved to `Next`

**Review by louisruch — APPROVED** [review]
One comment around the changelog entry

File: CHANGELOG.md (hunk 3 above)
Comment (jimlambrt) [reply CHANGELOG.md:19]: ty!

Comment (jimlambrt) [reply CHANGELOG.md]: ty!

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,15 +2,10 @@
 
 Canonical reference for changes, improvements, and bugfixes for Boundary.
 
-## Next
-
 ### New and Improved
 
-* config: Add support for go-sockaddr templates to Worker and Controller
```
Comment (louisruch) [inline CHANGELOG.md:9]: Seems you wiped the previous changelog entries and the `Next` title

Comment (jimlambrt) [reply CHANGELOG.md]: Oops.  Ty!

File: internal/servers/common/client_ip.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	"net/http"
+	"strings"
+	"sync/atomic"
+
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/go-secure-stdlib/listenerutil"
+)
+
+const (
+	RealIpHeader        = "X-Real-Ip"
```
Comment (jefferai) [inline internal/servers/common/client_ip.go:15]: Seems this isn't used anywhere right now?

**Review by jefferai — DISMISSED** [review]

File: internal/servers/common/client_ip.go (hunk 5 above)
Comment (jimlambrt) [reply internal/servers/common/client_ip.go:15]: ty!

**Review by louisruch — DISMISSED** [review]

**Review by jefferai — APPROVED** [review]

**Pull request merged**